### PR TITLE
Add prepare support in audio for alc5658 codec

### PR DESCRIPTION
**Pull request by btheosam**
Prepare helps to handle overflow and underflow errors in audio

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c:631: ERROR: [SPC_M_KWD] space required before the open brace '{'
apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c:631: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

File: os/drivers/audio/alc5658.c [hunk 1]
```diff
@@ -974,8 +961,35 @@ static int alc5658_ioctl(FAR struct audio_lowerhalf_s *dev, int cmd, unsigned lo
...
+		/* Get configured sample rate and set it here - valdiation done in configure itself */
+		entry = alc5658_get_sample_rate_script(priv->samprate);
+		alc5658_exec_i2c_script(priv, g_sample_entry[entry].script, g_sample_entry[entry].size / sizeof(t_codec_init_script_entry));
+
+		/* Set second set of registers */
+		alc5658_exec_i2c_script(priv, codec_init_inout_script2, sizeof(codec_init_inout_script2) / sizeof(t_codec_init_script_entry));
+
+		/* TOCHECK: Possible to cut the two level execution of alc scritps so as to cut the time? */
+		alc5658_setregs(priv);
+		alc5658_getregs(priv);
```
Comment (Taejun-Kwon) [inline os/drivers/audio/alc5658.c]: when it configured, getregs will be called.
Hence, information is enough. let's remove getregs here.

File: apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c [hunk 2]
```diff
@@ -628,6 +628,11 @@ static void utc_audio_pcm_readi_p(void)
 
 	while (remain > 0) {
 		frames_read = pcm_readi(g_pcm, buffer, pcm_get_buffer_size(g_pcm));
+		if(frames_read == -EPIPE){
+			pcm_prepare(g_pcm);
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c]: why don't you add pcm_prepare for writei function?
I think continue is ok for for readi, but writei is diffrent.
You can adjust do-while & TC_ASSERT sentence, I think

File: os/drivers/audio/alc5658.c (hunk 1 above)
Comment (btheosam) [reply os/drivers/audio/alc5658.c]: getregs will be executed only when audio logs, are enabled (for debugging purpose). static void alc5658_getregs(struct alc5658_dev_s *priv)
{
	audvdbg("MIC GAIN 0x%x\n", (uint32_t)alc5658_readreg(priv, ALC5658_IN1));
	audvdbg("MUTE HPOUT MUTE %x\n", (uint32_t)alc5658_readreg(priv, ALC5658_HPOUT));
	audvdbg("VOLL 0x%x\n", (uint32_t)alc5658_readreg(priv, ALC5658_HPOUT_L));
	audvdbg("VOLR 0x%x\n", (uint32_t)alc5658_readreg(priv, ALC5658_HPOUT_R));
}
NOw, the conriguration happens only in prepare ioclt.

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c:631: ERROR: [SPC_M_KWD] space required before the open brace '{'
apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c:631: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c:830: ERROR: [SPC_M_KWD] space required before the open brace '{'

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c:827: ERROR: [SPC_M_KWD] space required before the open brace '{'
apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c:827: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

File: apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c [hunk 3]
```diff
@@ -818,13 +823,20 @@ static void utc_audio_pcm_writei_p(void)
...
 
 	printf("Playback start!!\n");
-	do {
+
+	for(;;){
 		num_read = read(fd, buffer, size);
-		if (num_read > 0) {
-			ret = pcm_writei(g_pcm, buffer, pcm_bytes_to_frames(g_pcm, num_read));
-			TC_ASSERT_GEQ_CLEANUP("pcm_writei", ret, 0, clean_all_data(fd, buffer));
+		if (num_read <= 0) {
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c]: TO make this simply, just try writei one more time. please check below pseudo code.
```
ret = pcm_writei();
if (ret == -EPIPE) {
     pcm_prepare();
     ret = pcm_writei();
}
if (ret < 0) {
    TC_ASSERT();
}

File: apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c [hunk 4]
```diff
@@ -818,13 +823,20 @@ static void utc_audio_pcm_writei_p(void)
 	TC_ASSERT_GEQ_CLEANUP("pcm_writei", fd, 0, clean_all_data(0, buffer));
 
 	printf("Playback start!!\n");
-	do {
+
+	for(;;){
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c]: This brought coding style error, please check it.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**